configuration-api/main/advancedAuth.js

```diff
@@ -27,11 +36,28 @@ const typeDefs = gql`
     language: String!
     lastTenantSelected: String
   }
+  type ResourceType {
+    name: String!
+    userID: String!
+    tenantID: String!
+  }
+  type ResourceEndpoint {
+    name: String!
+    resourceTypeName: String!
+    nameAndID: String!
+  }
   type Query {
     listTenants(tenantNames: [String]!): [TenantConfiguration]
     getUserPreferences(userName: String!): [UserPreferencies]
+    getUserResourceType(tenantID: String!): [ResourceType]
+    getEndpoints(resourceTypeName: String!): [ResourceEndpoint]
   }
   type Mutation {
+    newResourceType(name: String!, userID: String!, tenantID: String!): [ResourceType]
+    deleteResourceType(name: [String]!): [ResourceType]
```

Comment:
same as above, different tenants may have the same name for a resource type. so you need define for which tenant you are deleting it.